src/color.rs

```diff
@@ -31,6 +31,18 @@ pub enum ColorType {
 
 }
 
+impl ColorType {
+    /// Returns the number of bits contained in a pixel of this `ColorType`
+    pub fn bits_per_pixel(&self) -> usize {
+        bits_per_pixel(*self)
+    }
+
+    /// Returns the number of color channels that make up this pixel
+    pub fn num_components(&self) -> usize {
```

Comment:
Same as the above but `u8` is maybe large enough already.

Reply:
Also, consider renaming to `channel_count` for naming alignment with the `Pixel` trait.